### DISCUSSION TITLE
Table widget

### DISCUSSION THREAD

**Discussion by Fee0** [hecrj/iced#1234] [Ideas]
Hi, need an advanced grid widget for my rust application. I want to show data in the grid and have features like swapping columns, changing column size, infinite scroll, etc. Iced seem to not over this out of the box. Would this be difficult to implement into iced? I'm rather new to developing new widgets for GUIs. And ideas?

**Comment by hecrj** [comment]
It's hard to grasp the scope of "an advanced `Grid` widget" by just mentioning some of the features!

What are some specific use cases you expect the widget to help you with? What is the simplest API we can come up with to tackle most of them?

By the sound of it, I believe some kind of `Table` widget is what you have in mind.

**Reply by Fee0** [reply]
Hi @hecrj, Thank you for your replay.
Yes, some kind of Table widget is what I mean. So I can visulize tabular data like an excel sheet and also do manipulations of the data and the layout of the table. ALso a kind of lazy loading of the data since the source data might be very big.
For an API its probably necessary to have at least, a data source (from which we derive dimensions and column titles or its provided as an  parameter), column ordering, table width and height, events for data changes, for lazy data loading a scroll position event, styles. And probably some more.
My requirements are not yet completly clear at the moment but this is how I imagine it right now.
How difficult is it to implement something like this in your opinion?

**Comment by ArekPiekarz** [comment]
Is it possible to create a table widget in the current Iced, like in a spreadsheet?

**Reply by hecrj** [reply]
There is an upcoming `iced_table` crate: https://github.com/tarkah/iced_table
